run_test.py

```diff
@@ -0,0 +1,229 @@
+    sys.exit(0)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument("-p", "--project")
+    parser.add_argument("-b", "--build", action="store_true")
+    parser.add_argument("-f", "--full", action="store_true")
+    parser.add_argument("-v", "--verbose", action="store_true")
+    parser.add_argument("-y", "--yaml", default="run_tests.yaml")
+    parser.add_argument("-h", "--help", action="store_true")
+    return parser.parse_args()
+
+
+def check_target():
+    if "TARGET" not in os.environ:
+        print("TARGET is not set.")
+        sys.exit(1)
+
+
+def read_yaml_config(yaml_path, project, full=False):
+    with open(yaml_path, "r") as file:
+        config = yaml.safe_load(file)
+
+    yaml_bin = "full_ci_dirs_bin" if full else "dirs_bin"
+    yaml_pytest = "full_ci_dirs_pytest" if full else "dirs_pytest"
+
+    search_dirs_bin = config[project][yaml_bin]
+    search_dirs_pytest = config[project][yaml_pytest]
+
+    return search_dirs_bin, search_dirs_pytest
+
+
+def change_directory(project):
+    if os.path.basename(os.getcwd()) != project:
+        if os.path.isdir(project):
+            os.chdir(project)
+        else:
+            os.chdir('..')
+
+
+def build_project():
+    print(f"{YELLOW} --- Start Building Project --- {NC}")
+    build_cmd = "./phoenix-rtos-build/build.sh clean all tests"
+    subprocess.run(build_cmd, shell=True, check=True)
+
+
+def prepare_environment(verbose):
+    print(f"{YELLOW} --- Prepare Environment --- {NC}")
+
+    test_dir = "test_por"
+    devtools_repo = "ssh://git.phoesys.com:29418/devtools"
+    devtools_dir = "devtools"
+
+    if os.path.isdir(test_dir):
+        print(f"Directory '{test_dir}' exists. Removing...")
+        subprocess.run(["rm", "-rf", test_dir], check=True)
+
+    if os.path.isdir(devtools_dir):
+        print(f"Directory '{devtools_dir}' exists. Removing...")
+        subprocess.run(["rm", "-rf", devtools_dir], check=True)
+
+    print(f"Cloning '{devtools_repo}' into '{devtools_dir}'...")
+    clone_cmd = ["git", "clone", devtools_repo, devtools_dir]
+    if verbose:
+        subprocess.run(clone_cmd, check=True)
+    else:
+        subprocess.run(clone_cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, check=True)
+
+    venv_dir = "_venv"
+    if os.path.isdir(venv_dir):
+        print(f"{YELLOW}Removing existing virtual environment '{venv_dir}'{NC}")
+        subprocess.run(["rm", "-rf", venv_dir], check=True)
+
+    print(f"{YELLOW}Creating new virtual environment '{venv_dir}'{NC}")
+
+    subprocess.run(["./utils/prepare-venv.py"], check=True)
+
+    for root, dirs, files in os.walk("."):
+        if "requirements.txt" in files:
+            requirements_path = os.path.join(root, "requirements.txt")
+            print(f"Installing dependencies from '{requirements_path}'")
+            try:
+                subprocess.run([os.path.join(venv_dir, "bin", "pip"), "install", "-r", requirements_path], check=True)
```

Comment:
I'm not sure if this approach is necessary or if it's great with subprocesses in a convenient manner. So I haven't included it so far because I didn't think it was crucial, but I will take a look to see if there might be a way to do this.